### DISCUSSION TITLE
trouble getting mimirtool working

### DISCUSSION THREAD

**Discussion by nwestatrussell** [grafana/mimir#7537] [Help and support] [answered]
I'm new to mimir and having a lot of trouble getting mimirtool working.
I've had the system admins create a service account for our tenants and from the Grafana UI generated a token.
To access the Grafana UI, I browse to "https://grafana.fqdn".

When I run
`mimirtool alertmanager get --auth-token="<token>" --address="https://grafana.fqdn/alerting" --id="<tenant id>" --log.level=debug`
I'm getting a 200 response code, but what is provided back is html, where the error implies it's expecting yaml.

When I run
`mimirtool alerts verify --auth-token="<token>" --address="https://grafana.fqdn" --id="<tenant id>" --log.level=debug`
I get a "404 Not Found"

When I try the same command, but with an http endpoint rather than https, 
I get a 200 response code, but the error
*mimirtool: error:  invalid character '<' looking for beginning of value*
which I assume means it's getting html back.

When I run
`mimirtool analyze ruler --address="https://grafana.fqdn" --id="<tenant id>" --log.level=debug --key="<token>"`
I get a "401 Unauthorized" "invalid username or password", despite using the same tenant id and token.

There are numerous other combinations of commands I've tried and associated errors, but I suspect there is a single underlying issue.
Any advice or suggestions on how to get mimirtool working would be much appreciated.

**Comment by charleskorn** [comment]
The value passed to `--address` should be the address of your Mimir cluster, not the address of your Grafana instance. If you're using the Helm chart, this would be the address of the nginx ingress.

**Reply by nwestatrussell** [reply]
Thanks charleskorn.
Switching to that address, I start getting 401 unauthorized, for all commands, presumably because the system account the token is generated against is a grafana account.  
I'll get the GEM admins to create an access policy for mimirtool and see how I go.

**Reply by dimitarvdimitrov** [reply]
if you're running GEM, you can also get more prompt support from GL by opening a ticket here https://grafana.com/profile/org/tickets/new (you should be logged in to your Grafana Labs account to which the GEM license was issued)

**Comment by tagathi** [comment, marked as answer]
Mimir itself has no service accounts and tokens. Those are Grafana related.
In order to use mimirtool you need to authenticate to Mimir.
I don't know how you have implemented authentication in Mimir, because Mimir itself does not offer authentication and authorization as described [here](https://grafana.com/docs/mimir/latest/manage/secure/authentication-and-authorization/).
